plugins/modules/cloudwatch_metric_alarm.py

```diff
@@ -204,18 +310,19 @@
     from botocore.exceptions import ClientError
 except ImportError:
     pass  # protected by AnsibleAWSModule
-
 from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible.module_utils.common.dict_transformations import camel_dict_to_snake_dict
+from ansible.module_utils.common.dict_transformations import snake_dict_to_camel_dict
 
 
 def create_metric_alarm(connection, module, params):
     alarms = connection.describe_alarms(AlarmNames=[params['AlarmName']])
-
-    if not isinstance(params['Dimensions'], list):
-        fixed_dimensions = []
-        for key, value in params['Dimensions'].items():
-            fixed_dimensions.append({'Name': key, 'Value': value})
-        params['Dimensions'] = fixed_dimensions
+    if params.get('Dimensions'):
+        if not isinstance(params['Dimensions'], list):
+            fixed_dimensions = []
+            for key, value in params['Dimensions'].items():
+                fixed_dimensions.append({'Name': key, 'Value': value})
+            params['Dimensions'] = fixed_dimensions
 
     if not alarms['MetricAlarms']:
         try:
```

Comment:
I know this wasn't part of this PR, but shouldn't the `changed = True` be outside this try/except block?

Reply:
yes. Only the API call should be inside the try/except block. Do you think we should make this fix in a separate PR, so that there will be clarity on what fixes are going in via a PR?

Reply:
Actually, this is fine. I think I read this the wrong way the first time and thought `changed` wouldn't be set to `True` in check mode. You can disregard.